### DISCUSSION TITLE
How does this work?

### DISCUSSION THREAD

**Discussion by AvtsVivek** [DeeJayTC/net-dynamic-api#15] [General]
Just curious, how does this work?

Does this use Roslyn source generators?

If yes, I dont see any generated files here.

![image](https://user-images.githubusercontent.com/41255281/161946058-55d09435-ef7d-4b33-8206-a3d82947fd8b.png)

**Comment by DeeJayTC** [comment]
Hey @AvtsVivek 
Its not using any code generators. The vnext will use some roslyn to generate stuff from JSON schema but other than that its all based on reflection, attributes and classes. 

The whole magic is happening at runtime. Heres a small post explaining a few things -> https://www.tcdev.de/instant-crud-apis-with-net

I'll write more about how things work later.

**Comment by DeeJayTC** [comment]
I'll turn this into a discussion to keep the issues clean
